src/Contexts/section.actions.jsx

```diff
@@ -0,0 +1,417 @@
+export const SAVING_SECTIONS = 'SAVING_SECTIONS'
+export const SAVING_SECTION_FAILED = 'SAVING_SECTION_FAILED'
+export const SECTION_SAVED = 'SECTION_SAVED'
+export const FETCHING_SECTION = 'FETCHING_SECTION'
+export const SECTION_FETCHED = 'SECTION_FETCHED'
+export const FETCHING_SECTION_FAILED = 'SECTION_FETCHED'
+
+/**
+ * Switch visibible of a section to true or false depending on the current value
+ * Will be synced with the server while save process and makes the sections
+ * invisible or visble for all how do not have write permissions
+ * 
+ * @param {string} sectionId - id of the sections witch visibility should switch
+ */
+export const switchSectionVisibility = (sectionId) => ({
+	type: SWITCH_SECTION_VISIBILTY,
+	payload: sectionId
+})
+
+/**
+ * Set Sections in sotre, if there are existing sections they will overwirden
+ * to add a Section use addSection
+ * 
+ * @param {Object[]} sections - Sections to overwrite current sections
+ */
+export const setSections = (sections) => ({
+	type: SET_SECTIONS,
+	payload: sections
+})
+
+/**
+ * Add a section to the current state of sections, it is recommend to set an Id at least
+ * But could also called with out any parameter. Sections will be set with following parametes
+ * if no other value is given
+ * 
+ * Defaults:
+ * - title: ""
+ * - _id: uuid - should be set to a mongoDB id, otherwise it could not saved to server
+ * - visible: true - also read users can see the section
+ * - docValue: {} - state object of serlo editor
+ * - changed: Set - have to be an Set Obejct with last changes, shouldn't be overwrite
+ * - position: -1 - section will added at least positon
+ * 
+ * @param {Object} [section] - section object, all necesarry data are setten by default and can be overwirten
+ */
+export const addSection = (section) => ({
+	type: ADD_SECTION,
+	payload: {
+		title: "",
+		_id: uuid(), // TODO: mark, so it will not be saved to backend with this id
+		visible: true,
+		docValue: {},
+		changed: new Set(),
+		position: -1, // added at last pos
+		...section
+	}
+})
+
+/**
+ * Fetch one or multible sections from the server and add it to sections state
+ * if no positon is setted by the server response it will be added at last position
+ *
+ * @param  {...String} sectionIds - comma seperated list of section ids
+ */
+export const fetchSection = (...sectionIds) => ({state, dispatch}) => {
+
+	// TODO: check connection and inform user
+
+	const proms = sectionIds.map(async (_id) => {
+
+		dispatch({
+			type: FETCHING_SECTION,
+			payload: _id
+		})
+
+		return editorWS.emit(
+			'get',
+			`lesson/${state.lesson._id}/sections`,
+			_id
+		)
+	})
+
+	const resolved = Promise.allSettled(proms)
+	resolved.forEach((res, i) => {
+		if(res.status === 'fulfilled'){
+			dispatch(addSection(res.value))
+			dispatch({
+				type: SECTION_FETCHED,
+				payload: res.value._id
+			})
+		} else {
+			dispatch({
+				type: FETCHING_SECTION_FAILED,
+				payload: sectionIds[i]
```

Comment:
Ist der index eine sicherer Methode um die Position der Section zu bestimmen? Diese ist doch aktuell nur in der lesson referenziert.